app/controllers/TwoEyeReviewResultController.scala

```diff
@@ -44,17 +43,32 @@ class TwoEyeReviewResultController @Inject() (
     twoEyeReviewerIdentifierAction: TwoEyeReviewerIdentifierAction,
     formProvider: TwoEyeReviewResultFormProvider,
     view: twoeye_review_result,
+    errorView: twoeye_confirm_error,
     reviewService: ReviewService,
     mcc: MessagesControllerComponents
 ) extends FrontendController(mcc)
     with I18nSupport {
 
-  val logger = Logger(getClass)
+  val logger: Logger = Logger(getClass)
 
   def onPageLoad(processId: String): Action[AnyContent] = twoEyeReviewerIdentifierAction.async { implicit request =>
-    val form: Form[ApprovalStatus] = formProvider()
 
-    Future.successful(Ok(view(processId, form)))
+    reviewService.approval2iReviewCheck(processId) map {
+      case Right(()) =>
+        val form: Form[ApprovalStatus] = formProvider()
+        Ok(view(processId, form))
+      case Left(IncompleteDataError) => Ok(errorView(processId))
+      case Left(NotFoundError) =>
+        logger.error(s"Unable to check 2i review approval for process $processId not found")
+        NotFound(errorHandler.notFoundTemplate)
+      case Left(StaleDataError) =>
+        logger.warn(s"The requested 2i review approval check for process $processId failed - stale data")
+        NotFound(errorHandler.notFoundTemplate)
+      case Left(err) =>
+        // Handle stale data, internal server and any unexpected errors
```

Comment:
Stale data is no longer needed in comment

Reply:
done, included in next commit